tests/MongoDB.Driver.Tests/PinnedShardRouterTests.cs

```diff
@@ -176,7 +176,8 @@ private DisposableMongoClient CreateDisposableClient(EventCapturer eventCapturer
                 }, 
                 useMultipleShardRouters);
             var timeOut = TimeSpan.FromSeconds(60);
-            SpinWait.SpinUntil(() => client.Cluster.Description.Type != ClusterType.Unknown, timeOut).Should().BeTrue();
+            bool AllServersConnected() => client.Cluster.Description.Servers.All(s => s.State == ServerState.Connected);
```

Comment:
as our internal agreement, we place local functions in the end of method :)

Reply:
I think we need to mull that over some more.

In this case it feels more natural to place it right before first use.

Maybe the idea should be treat local functions like variables (even though strictly speaking they are not variables).

Reply:
idk, I don't see a big difference between placing it in line 179 or 182, it's still close. 
My point that I would prefer to have as more as possible explicit rules to avoid unnecessary debates on them further :)
In general I don't mind to have it in line 179, but only if we apply this rule for all other places.

Reply:
If reviewers insist on moving this local function to the end of the method and capitalizing it then I revise my suggestion to use the following instead:

    Func<bool> allServersConnected() => client.Cluster.Description.Servers.All(s => s.State == ServerState.Connected);
    SpinWait.SpinUntil(allServersConnected, timeOut).Should().BeTrue();

Reply:
>but only if we apply this rule for all other places.
at least it sounds reasonable for me. Otherwise, we can have potential disagreements about it later. Which is probably not necessary for this simple case

Reply:
Since we have two lgtms, I think I will push this as is and we can hash out the style another day perhaps :) Thank you Dima and Robert for sharing your thoughts on this though! Style questions are certainly tricky :)